allen-cell-animated/nucmorph-colorizer
# feature: Save UI settings to URL

Problem
=======
Closes #183, saving UI settings to the URL.

*Expected review time: medium, <30 min*

Solution
========
- Adds methods to serialize and deserialize the `ViewerConfig`.
- Adds `config (ViewerConfig)` and `selectedBackdropKey` to the URL.
- Adds unit tests.

## Type of change
* New feature (non-breaking change which adds functionality)


Steps to Verify:
----------------
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/47125ec8-b83e-434b-93ec-bc2477d56c5d)


1. Click the following link and open the preview: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-219/?dataset=https%3A%2F%2Fdev-aics-dtp-001.int.allencell.org%2Fdan-data%2Fcolorizer%2Fdata%2Ftest%2F3500005828_25%2Fmanifest.json&feature=Nuclei%20area&t=84&filters=Radial%2520distance%2520from%2520colony%2520center%3A%3A0.250%3A180.250&color=matplotlib-cool&palette-key=adobe&bg-sat=40&bg-brightness=90&fg-alpha=61&outlier-color=c0c0c0&outlier-mode=0&filter-color=adff1a&filter-mode=1&scalebar=0&timestamp=0&path=0&keep-range=0&bg-key=ch0ch1
1. You should see matching settings if you open the Settings tab.
2. (Note: If you see the first frame, this is a bug that's currently captured by #171. Try reopening the link.)
